docs/configuration-ini.md

```diff
@@ -344,71 +589,97 @@ activation = softmax
 ```
 
 ### Configuration file example
+Only INI formatted files *.ini is supported to construct a model from a file.
+Special sections [Model], [Optimizers], [LearningRateScheduler], [train_set], [valid_set], [test_set] are respectively referring to model, optimizer and data provider objects. Rest of INI sections map to a layer. Keys and values from each section set properties of the layer. All keys and values are treated as case-insensitive.
+
+The following restrictions must be adhered to:
 
-This has one input layer, two convolution layers, two pooling layers, one flatten layer and one fully connected layer to classify MNIST example.
+ - Model file must have a `[Model]` section.
+ - Model file must have at least one layer.
+ - Valid keys must have valid properties. The invalid keys in each section result in an error.
+ - All paths inside the INI file are relative to the INI file path unless the absolute path is stated.
 
+Below is sample backbone section.
 It takes 1 x 28 x 28 gray data (0~255) as an input. Adam optimizer is used to apply gradient and learning rate is 1.0e-4.
 
 ```ini
 # Model Section
 [Model]
-type = NeuralNetwork
-learning_rate = 1e-4
-epochs = 1500
-optimizer = adam
-loss = cross
-Save_Path = "model.bin"
-batch_size = 32
-beta1 = 0.9
-beta2 = 0.999
-epsilon = 1e-7
+Type = NeuralNetwork          # Network Type : Regression, KNN, NeuralNetwork
+Epochs = 1500                 # Epochs
+Loss = cross                  # Loss function : mse (mean squared error)
+                              #                 cross ( for cross entropy )
+Save_Path = "mnist_model.bin" # model path to save / read
+batch_size = 32               # batch size
+
+[Optimizer]
+Type = adam
+beta1 = 0.9       # beta 1 for adam
+beta2 = 0.999     # beta 2 for adam
+epsilon = 1e-7    # epsilon for adam
+
+[LearningRateScheduler]
+type=constant
+Learning_rate = 1e-4 # Learning Rate
+
+# Train Set Section
+[train_set]
+Type = file
+path = "trainDataset.dat"
 
-# Layer Section
+# Layer Section : Name
 [inputlayer]
-type = input
-input_shape = 1:28:28
+Type = input
+Input_Shape = 1:28:28
 
+# Layer Section : Name
 [conv2d_c1_layer]
-type = conv2d
+Type = conv2d
+input_layers = inputlayer
 kernel_size = 5,5
 bias_initializer=zeros
-activation=sigmoid
+Activation=sigmoid
```

Comment:
Just for the information, ini file is not type sensitive.

Reply:
okay